src/Assertions.jl

```diff
@@ -0,0 +1,45 @@
+########################################################################
+# Internal sanity checks for input
+#
+# Many functions in Oscar make use of an optional argument 
+# `check::Bool=true`. If enabled, the function will run expensive 
+# sanity checks on the input using the `@check` macro.
+#
+# For internal calls to such functions with input which has already 
+# been verified, one should then set this argument to `false`. However, 
+# this is often forgotten when programming. In order to systematically 
+# snoop for such leaks, the following global variable can be set to `true` 
+# and then tests can be run which are supposed to *not* trigger any such 
+# expensive input checks. If an internal test is nevertheless run with 
+# @check, then an error is thrown and the faulty call can be disabled 
+# following the subsequent stacktrace. 
+########################################################################
+
+global const THROW_ERROR_FOR_INTERNAL_CHECKS = false
+
+macro check()
+  quote
+    if $(esc(:(check)))
+      THROW_ERROR_FOR_INTERNAL_CHECKS && error("internal checks are enabled")
+    end
+  end
+end
+
+macro check(cond)
+  quote
+    if $(esc(:(check)))
+      THROW_ERROR_FOR_INTERNAL_CHECKS && error("internal checks are enabled")
+      @assert $(esc(cond))
```

Comment:
But `@assert` can be turned off; yet in the next macro, you use
```
$(esc(cond)) || error($(esc(msg)))
```
which cannot be turned off like other assertions. I don't think users of the `@check` macro would expect that adding a message turns it from an assertion into a non-assertion...

So either both should use `@assert` (the second macro then in the two argument form `@assert cond msg`, or both should not use `@assert` (and thus *always* evaluate `cond`. I think the latter is more appropriate here, isn't it?